### PR TITLE
[mono][llvm] Fix an llvm assertion.

### LINKED ISSUE #97217: [browser][AOT] convert the LLVM value V to DTYPE
### Error Blob

```json
{
  "ErrorMessage": "",
  "BuildRetry": false,
  "ErrorPattern": "Assertion: should not be reached at .*mini-llvm.c:1394",
  "ExcludeConsoleLog": false
}
```

### Reproduction Steps

[Log](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-96841-merge-0fea1a5b978548eb82/WasmTestOnBrowser-System.Runtime.Intrinsics.Tests/1/console.4ffe3338.log?helixlogtype=result)
[Log](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-96841-merge-f085d69333c546e49c/WasmTestOnBrowser-System.Runtime.Intrinsics.Tests/1/console.a70fb307.log?helixlogtype=result)
```
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(675,5): error : Precompiling failed for C:\helix\work\workitem\e\wasm_build\obj\wasm\for-build\aot-in\System.Private.CoreLib.dll with exit code -1073740791. [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(675,5): error : Mono Ahead of Time compiler - compiling assembly C:\helix\work\workitem\e\wasm_build\obj\wasm\for-build\aot-in\System.Private.CoreLib.dll [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(675,5): 
error :   %process_call_vtype_ret_addr = load <1 x double>, ptr %vreg_loc_23, align 8 [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(675,5): 
error : <2 x double> [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(675,5): 
error : * Assertion: should not be reached at 
D:\a\_work\1\s\src\mono\mono\mini\mini-llvm.c:1394 [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
```

Build: https://dev.azure.com/dnceng-public/public/_build/results?buildId=533238

 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=533238
**Error message validated:** `Assertion: should not be reached at .*mini-llvm.c:1394`
**Result validation:** :white_check_mark: Known issue matched with the provided build.
**Validation performed at:** 1/19/2024 3:17:34 PM UTC

### Report

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[537134](https://dev.azure.com/dnceng-public/public/_build/results?buildId=537134)|dotnet/runtime|[WasmTestOnBrowser-System.Runtime.Intrinsics.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=537134&view=ms.vss-test-web.build-test-results-tab&runId=12703250&resultId=166968)|dotnet/runtime#97270|
|[536976](https://dev.azure.com/dnceng-public/public/_build/results?buildId=536976)|dotnet/runtime|[WasmTestOnBrowser-System.Runtime.Intrinsics.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=536976&view=ms.vss-test-web.build-test-results-tab&runId=12696576&resultId=168819)|dotnet/runtime#97299|
|[2359804](https://dev.azure.com/dnceng/internal/_build/results?buildId=2359804)|dotnet-runtime|[release.Partition9.WorkItemExecution](https://dev.azure.com/dnceng//internal/_build/results?buildId=2359804&view=ms.vss-test-web.build-test-results-tab&runId=52928661&resultId=100030)||
|[2359653](https://dev.azure.com/dnceng/internal/_build/results?buildId=2359653)|dotnet-runtime|[x64.micro_mono.net9.0.Partition9.WorkItemExecution](https://dev.azure.com/dnceng//internal/_build/results?buildId=2359653&view=ms.vss-test-web.build-test-results-tab&runId=52927973&resultId=100030)||
|[536434](https://dev.azure.com/dnceng-public/public/_build/results?buildId=536434)|dotnet/runtime|[WasmTestOnBrowser-System.Runtime.Intrinsics.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=536434&view=ms.vss-test-web.build-test-results-tab&runId=12681782&resultId=168819)|dotnet/runtime#97299|
|[2359386](https://dev.azure.com/dnceng/internal/_build/results?buildId=2359386)|dotnet-runtime|[release.Partition9.WorkItemExecution](https://dev.azure.com/dnceng//internal/_build/results?buildId=2359386&view=ms.vss-test-web.build-test-results-tab&runId=52926751&resultId=100030)||
|[536247](https://dev.azure.com/dnceng-public/public/_build/results?buildId=536247)|dotnet/runtime|[WasmTestOnBrowser-System.Runtime.Intrinsics.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=536247&view=ms.vss-test-web.build-test-results-tab&runId=12678604&resultId=168827)|dotnet/runtime#97270|
|[535842](https://dev.azure.com/dnceng-public/public/_build/results?buildId=535842)|dotnet/runtime|[WasmTestOnBrowser-System.Runtime.Intrinsics.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=535842&view=ms.vss-test-web.build-test-results-tab&runId=12665934&resultId=168827)|dotnet/runtime#97270|
|[2359131](https://dev.azure.com/dnceng/internal/_build/results?buildId=2359131)|dotnet-runtime|[x64.micro_mono.net9.0.Partition9.WorkItemExecution](https://dev.azure.com/dnceng//internal/_build/results?buildId=2359131&view=ms.vss-test-web.build-test-results-tab&runId=52923919&resultId=100030)||
|[535620](https://dev.azure.com/dnceng-public/public/_build/results?buildId=535620)|dotnet/runtime|[WasmTestOnBrowser-System.Runtime.Intrinsics.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=535620&view=ms.vss-test-web.build-test-results-tab&runId=12659984&resultId=168823)|dotnet/runtime#97270|
|[535319](https://dev.azure.com/dnceng-public/public/_build/results?buildId=535319)|dotnet/runtime|[WasmTestOnBrowser-System.Runtime.Intrinsics.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=535319&view=ms.vss-test-web.build-test-results-tab&runId=12650782&resultId=168823)|dotnet/runtime#97270|
|[2358898](https://dev.azure.com/dnceng/internal/_build/results?buildId=2358898)|dotnet-runtime|[release.Partition9.WorkItemExecution](https://dev.azure.com/dnceng//internal/_build/results?buildId=2358898&view=ms.vss-test-web.build-test-results-tab&runId=52921787&resultId=100030)||
|[535175](https://dev.azure.com/dnceng-public/public/_build/results?buildId=535175)|dotnet/runtime|[WasmTestOnBrowser-System.Runtime.Intrinsics.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=535175&view=ms.vss-test-web.build-test-results-tab&runId=12644538&resultId=168823)|dotnet/runtime#97256|
|[2358750](https://dev.azure.com/dnceng/internal/_build/results?buildId=2358750)|dotnet-runtime|[release.Partition9.WorkItemExecution](https://dev.azure.com/dnceng//internal/_build/results?buildId=2358750&view=ms.vss-test-web.build-test-results-tab&runId=52919191&resultId=100030)||
|[535008](https://dev.azure.com/dnceng-public/public/_build/results?buildId=535008)|dotnet/runtime|[WasmTestOnBrowser-System.Runtime.Intrinsics.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=535008&view=ms.vss-test-web.build-test-results-tab&runId=12637406&resultId=168823)|dotnet/runtime#97270|
|[534976](https://dev.azure.com/dnceng-public/public/_build/results?buildId=534976)|dotnet/runtime|[WasmTestOnBrowser-System.Runtime.Intrinsics.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=534976&view=ms.vss-test-web.build-test-results-tab&runId=12635986&resultId=168823)|dotnet/runtime#97270|
|[2358610](https://dev.azure.com/dnceng/internal/_build/results?buildId=2358610)|dotnet-runtime|[release.Partition9.WorkItemExecution](https://dev.azure.com/dnceng//internal/_build/results?buildId=2358610&view=ms.vss-test-web.build-test-results-tab&runId=52918315&resultId=100030)||
|[2358539](https://dev.azure.com/dnceng/internal/_build/results?buildId=2358539)|dotnet-runtime|[release.Partition9.WorkItemExecution](https://dev.azure.com/dnceng//internal/_build/results?buildId=2358539&view=ms.vss-test-web.build-test-results-tab&runId=52917959&resultId=100030)||
|[534811](https://dev.azure.com/dnceng-public/public/_build/results?buildId=534811)|dotnet/runtime|[WasmTestOnBrowser-System.Runtime.Intrinsics.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=534811&view=ms.vss-test-web.build-test-results-tab&runId=12632036&resultId=166964)|dotnet/runtime#97270|
|[2358482](https://dev.azure.com/dnceng/internal/_build/results?buildId=2358482)|dotnet-runtime|[x64.micro_mono.net9.0.Partition9.WorkItemExecution](https://dev.azure.com/dnceng//internal/_build/results?buildId=2358482&view=ms.vss-test-web.build-test-results-tab&runId=52917635&resultId=100030)||
|[2358409](https://dev.azure.com/dnceng/internal/_build/results?buildId=2358409)|dotnet-runtime|[x64.micro_mono.net9.0.Partition9.WorkItemExecution](https://dev.azure.com/dnceng//internal/_build/results?buildId=2358409&view=ms.vss-test-web.build-test-results-tab&runId=52916795&resultId=100030)||
|[2358317](https://dev.azure.com/dnceng/internal/_build/results?buildId=2358317)|dotnet-runtime|[release.Partition9.WorkItemExecution](https://dev.azure.com/dnceng//internal/_build/results?buildId=2358317&view=ms.vss-test-web.build-test-results-tab&runId=52916057&resultId=100030)||
|[534555](https://dev.azure.com/dnceng-public/public/_build/results?buildId=534555)|dotnet/runtime|[WasmTestOnBrowser-System.Runtime.Intrinsics.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=534555&view=ms.vss-test-web.build-test-results-tab&runId=12622414&resultId=166964)|dotnet/runtime#95370|
|[2358234](https://dev.azure.com/dnceng/internal/_build/results?buildId=2358234)|dotnet-runtime|[x64.micro_mono.net9.0.Partition9.WorkItemExecution](https://dev.azure.com/dnceng//internal/_build/results?buildId=2358234&view=ms.vss-test-web.build-test-results-tab&runId=52915329&resultId=100030)||
|[534349](https://dev.azure.com/dnceng-public/public/_build/results?buildId=534349)|dotnet/runtime|[WasmTestOnBrowser-System.Runtime.Intrinsics.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=534349&view=ms.vss-test-web.build-test-results-tab&runId=12616202&resultId=166954)|dotnet/runtime#95370|
|[2358104](https://dev.azure.com/dnceng/internal/_build/results?buildId=2358104)|dotnet-runtime|[release.Partition9.WorkItemExecution](https://dev.azure.com/dnceng//internal/_build/results?buildId=2358104&view=ms.vss-test-web.build-test-results-tab&runId=52913679&resultId=100030)||
|[2357872](https://dev.azure.com/dnceng/internal/_build/results?buildId=2357872)|dotnet-runtime|[x64.micro_mono.net9.0.Partition9.WorkItemExecution](https://dev.azure.com/dnceng//internal/_build/results?buildId=2357872&view=ms.vss-test-web.build-test-results-tab&runId=52911415&resultId=100030)||
|[533712](https://dev.azure.com/dnceng-public/public/_build/results?buildId=533712)|dotnet/runtime|[WasmTestOnBrowser-System.Runtime.Intrinsics.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=533712&view=ms.vss-test-web.build-test-results-tab&runId=12599912&resultId=168813)|dotnet/runtime#95370|
|[2357620](https://dev.azure.com/dnceng/internal/_build/results?buildId=2357620)|dotnet-runtime|[release.Partition9.WorkItemExecution](https://dev.azure.com/dnceng//internal/_build/results?buildId=2357620&view=ms.vss-test-web.build-test-results-tab&runId=52909449&resultId=100030)||
|[533564](https://dev.azure.com/dnceng-public/public/_build/results?buildId=533564)|dotnet/runtime|[WasmTestOnBrowser-System.Runtime.Intrinsics.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=533564&view=ms.vss-test-web.build-test-results-tab&runId=12596418&resultId=166954)|dotnet/runtime#96354|
|[533238](https://dev.azure.com/dnceng-public/public/_build/results?buildId=533238)|dotnet/runtime|[WasmTestOnBrowser-System.Runtime.Intrinsics.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=533238&view=ms.vss-test-web.build-test-results-tab&runId=12587854&resultId=166955)|dotnet/runtime#96841|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|9|31|31|

### PR DESCRIPTION
**Pull request by vargaz**
Fixes https://github.com/dotnet/runtime/issues/97217.

### CODE REVIEW

File: src/mono/mono/mini/simd-intrinsics.c [hunk 1]
```diff
@@ -1712,6 +1712,18 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 			if (!COMPILE_LLVM (cfg))
 				return NULL;
 #endif
+			if (COMPILE_LLVM (cfg)) {
+				/*
+				 * The sregs are half size, and the dreg is full size
+				 * which can cause problems if mono_handle_global_dregs () is trying to
```
Comment (lambdageek) [inline src/mono/mono/mini/simd-intrinsics.c:1718]:
```suggestion
				 * which can cause problems if mono_handle_global_vregs () is trying to
```

**Review by lambdageek — APPROVED** [review]

**Comment by vargaz** [comment]
Failures/timeouts appear unrelated.

**Pull request merged**